ts/sagas/wallet.ts

```diff
@@ -636,11 +631,29 @@ export function* watchWalletSaga(
     pagopaNodoClient.postAttivaRpt
   );
 
-  yield takeLatest(
-    getType(paymentIdPolling.request),
-    paymentIdPollingRequestHandler,
-    pollingPagopaNodoClient.getPaymentId
-  );
+  yield takeLatest(getType(paymentIdPolling.request), function*(
+    action: ActionType<typeof paymentIdPolling["request"]>
+  ) {
+    shouldAbortPaymentIdPollingRequest = DeferredPromise<boolean>();
+    // Backend client for polling for paymentId - uses an instance of fetch that
+    // considers a 404 as a transient error and retries with a constant delay
+    // it is created every time paymentIdPolling.request is triggered:
+    // this is because we have to re-new shouldAbortPaymentIdPollingRequest to be able to abort the running polling
+    const pollingPagopaNodoClient = BackendClient(
```

Comment:
another option is to create the deffered promise and return it inside the fetch call